zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -241,8 +257,40 @@ def get_start_request(self, url):
         )
 
     def start_requests(self) -> Iterable[Request]:
-        for url in self.start_urls:
-            yield self.get_start_request(url)
+        if self.args.search_queries:
+            for url in self.start_urls:
+                meta: Dict[str, Any] = {
+                    "crawling_logs": {"page_type": "searchRequestTemplate"},
+                }
+                if self.args.crawl_strategy == EcommerceCrawlStrategy.full:
+                    meta["page_params"] = {"full_domain": get_domain(url)}
+                if self.args.extract_from == ExtractFrom.browserHtml:
+                    meta["inject"] = [BrowserResponse]
+                yield Request(
+                    url=url,
+                    callback=self.parse_search_request_template,
+                    meta=meta,
+                )
+        else:
+            for url in self.start_urls:
+                yield self.get_start_request(url)
+
+    def parse_search_request_template(
+        self,
+        response: DummyResponse,
+        search_request_template: SearchRequestTemplate,
+        dynamic: DynamicDeps,
+    ) -> Iterable[Request]:
+        probability = search_request_template.get_probability()
+        if probability is not None and probability <= 0:
+            return
+        for query in self.args.search_queries:
+            yield search_request_template.request(query=query).to_scrapy(
+                callback=self.parse_navigation,
```

Comment:
Hm, I wonder if we should disable following subCategories on the target pages. When checking how it works, did you have cases where it'd be helpful to follow subcategories? 

If they're not useful in practice, following them can be harmful, as it's possible that the ML model will extract something which is not subCategory - this would lead to extraction of items unrelated to the search queries.

So, the question is - what is more frequent - cases with subcategories on search result pages, or cases where on search result pages subcategories are detected, but they're wrong?